test/integration_test/extender_test.go

```diff
@@ -146,3 +148,52 @@ func driverNodeErrorTest(t *testing.T) {
 
 	destroyAndWait(t, ctxs)
 }
+
+func pvcOwnershipTest(t *testing.T) {
+	ctxs, err := schedulerDriver.Schedule(generateInstanceID(t, "pvcownershiptest"),
+		scheduler.ScheduleOptions{AppKeys: []string{"mysql-1-pvc"}})
+	require.NoError(t, err, "Error scheduling task")
+	require.Equal(t, 1, len(ctxs), "Only one task should have started")
+
+	err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "Error waiting for pod to get to running state")
+
+	scheduledNodes, err := schedulerDriver.GetNodesForApp(ctxs[0])
+	require.NoError(t, err, "Error getting node for app")
+	require.Equal(t, 1, len(scheduledNodes), "App should be scheduled on one node")
+
+	volumeNames := getVolumeNames(t, ctxs[0])
+	require.Equal(t, 1, len(volumeNames), "Should have only one volume")
+
+	verifyScheduledNode(t, scheduledNodes[0], volumeNames)
+
+	for _, spec := range ctx.App.SpecList {
+		if obj, ok := spec.(*storage_api.StorageClass); ok {
+			if s, err := k8s.Instance().DeleteStorageClass(obj.Name); err != nil {
+				if err != nil {
+					return err, &scheduler.ErrFailedToDestroyStorage{
+						App:   ctx.App,
+						Cause: fmt.Sprintf("Failed to destroy storage class: %v. Err: %v", obj.Name, err),
+					}
+				}
+			}
+		}
+	}
+
+	for _, spec := range ctx.App.SpecList {
+		if obj, ok := spec.(*vi.PersistentVolumeClaim); ok {
+			updatePVC, _ := k8sOps.GetPersistentVolumeClaim(obj.Name, obj.Namespace)
+			if annotationValue, hasKey := updatePVC.Annotations["storage-provisioner"]; hasKey {
+				delete(updatePVC, "storage-provisioner")
```

Comment:
Need to call UpdatePersistentVolumeClaim() after updating the annotation

Reply:
Fixed.